File: Core/Entities/Coupon.cs
namespace Core.Entities
{
    public class Coupon : BaseEntity
    {
        public required string Code { get; set; }
        public required string Description { get; set; }
        public CouponType Type { get; set; }
        public decimal Value { get; set; } // Percentage or fixed amount
        public decimal? MinimumOrderAmount { get; set; }
        public decimal? MaximumDiscountAmount { get; set; } // For percentage coupons
        public DateTime ValidFrom { get; set; }
        public DateTime ValidUntil { get; set; }
        public int? UsageLimit { get; set; } // Null for unlimited
        public int UsageCount { get; set; } = 0;
        public bool IsActive { get; set; } = true;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public bool IsCustomerOnly { get; set; } = false; // Only for regular customers (not company users)

        // Navigation property
        public ICollection<CouponUsage> CouponUsages { get; set; } = [];
    }

    public enum CouponType
    {
        Percentage = 1,
        FixedAmount = 2
    }
}

File: client/src/app/shared/models/coupon.ts
export type Coupon = {
  id: number;
  code: string;
  description: string;
  type: string;
  value: number;
  minimumOrderAmount?: number;
  maximumDiscountAmount?: number;
  validFrom: string;
  validUntil: string;
  usageLimit?: number;
  usageCount: number;
  isActive: boolean;
  createdAt: string;
  isCustomerOnly: boolean;
}

export type CreateCouponDto = {
  code: string;
  description: string;
  type: number;
  value: number;
  minimumOrderAmount?: number;
  maximumDiscountAmount?: number;
  validFrom: string;
  validUntil: string;
  usageLimit?: number;
  isCustomerOnly: boolean;
}

export type CouponType = {
  value: number;
  name: string;
}

export const CouponTypes: CouponType[] = [
  { value: 1, name: 'Percentage' },
  { value: 2, name: 'Fixed Amount' }
];